rki-mf1/covsonar
# Prepare packing and automated testing

Reorganize the repo and set up the necessary files to package the software using poetry, do linting and testing with nox, and easily reformat the code and imports with black and zimports. Added some basic tests using pytest, but the "import" task is currently untested because it is causing crashes.

See CONTRIBUTING.md for the info you need to get started. From the TLDR section of that document:

```
mamba create -n covsonar-dev python=3.9 poetry fortran-compiler nox pre-commit emboss=6.6.0
mamba activate covsonar-dev  # needs to be activated for the following commands to work
poetry install  # install current source of covsonar and its dependencies
poetry run covsonar <args>  # run covsonar
nox  # run linting and pytest tests
nox -s zimports black  # auto format imports and code
```

Still to do:
- publish the alpha packages to testpypi.org and pypi.org, and use the latter to create a bioconda package
- figure out the "import" testing crash
- set up pre-commit tests to enforce formatting and linting
- ensure data files are properly included in the published packages (reference genome, etc.)
